building with LibreSSL 3.9.0 and later fails because EVP_add_cipher_alias() no longer exists

```
$ make -s CFLAGS=-Werror
print-esp.c: In function 'esp_init':
print-esp.c:681:9: error: implicit declaration of function 'EVP_add_cipher_alias' [-Werror=implicit-function-declaration]
  681 |         EVP_add_cipher_alias(SN_des_ede3_cbc, "3des");
      |         ^~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
*** Error 1 in /home/buildbot-worker/openbsd-amd64/tmpfs/tcpdump-openbsd-amd64/build (Makefile:68 'print-esp.o': egcc -W -Wall -Wcast-qual -...)
*** Error 1 in target 'print-esp.o'
```
In OpenBSD 7.4 [the header](https://cvsweb.openbsd.org/src/lib/libcrypto/evp/evp.h) defines `EVP_add_cipher_alias()` as follows:
```c
#define EVP_add_cipher_alias(n,alias) \
        OBJ_NAME_add((alias),OBJ_NAME_TYPE_CIPHER_METH|OBJ_NAME_ALIAS,(n))
```
In OpenBSD 7.5 it does not define the macro since revision 1.122 (2024-03-02). One way to resolve this would be to make the alias conditional, another would be to stop using it (it was introduced in commit 0d821a5).

Sure, 

```
<simonb@odin:~/src/bpf/ping>
zsh/2 1221 % openssl version
LibreSSL 3.9.2
```

Usual configure:
```
./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var

```
Output snipped:
```...
checking whether to use OpenSSL/libressl libcrypto... yes, if available
checking openssl/crypto.h usability... yes
checking openssl/crypto.h presence... yes
checking for openssl/crypto.h... yes
checking for DES_cbc_encrypt in -lcrypto... yes
checking openssl/evp.h usability... yes
checking openssl/evp.h presence... yes
checking for openssl/evp.h... yes
checking for EVP_CIPHER_CTX_new... yes
checking for EVP_DecryptInit_ex... yes
...
```

A warning is emitted during make, the build continues though:
```
gcc  -DHAVE_CONFIG_H   -I. -I/usr/include/libnl3 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include  -g -O2 -c ./print-esp.c
./print-esp.c: In function 'esp_init':
./print-esp.c:686:9: warning: implicit declaration of function 'EVP_add_cipher_alias' [-Wimplicit-function-declaration]
  686 |         EVP_add_cipher_alias(SN_des_ede3_cbc, "3des");
      |         ^~~~~~~~~~~~~~~~~~~~
```

and finally:
```
ranlib libnetdissect.a
gcc  -DHAVE_CONFIG_H   -I. -I/usr/include/libnl3 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include  -g -O2  -o tcpdump fptype.o tcpdump.o libnetdissect.a -lcap-ng -lcrypto -lpcap 
/usr/bin/ld: libnetdissect.a(print-esp.o): in function `esp_init':
/usr/src/misc/tcpdump-4.99.4/./print-esp.c:686:(.text+0xe26): undefined reference to `EVP_add_cipher_alias'
collect2: error: ld returned 1 exit status
make: *** [Makefile:395: tcpdump] Error 1
```

Which is hardly surprising, as:
```
<simonb@odin:/usr/src/misc/tcpdump-4.99.4>
zsh/2 1233 [1] % nm /usr/lib/libcrypto.so | grep EVP_add_cipher_alias || echo notfound
notfound
<simonb@odin:/usr/src/misc/tcpdump-4.99.4>
zsh/2 1234 % nm /usr/lib/libssl.so | grep EVP_add_cipher_alias || echo notfound
notfound
```

--- a/print-esp.c
+++ b/print-esp.c
@@ -425,6 +425,7 @@ espprint_decode_encalgo(netdissect_options *ndo,
 	const EVP_CIPHER *evp;
 	int authlen = 0;
 	char *colon, *p;
+	const char *real_decode;
 
 	colon = strchr(decode, ':');
 	if (colon == NULL) {
@@ -445,10 +446,23 @@ espprint_decode_encalgo(netdissect_options *ndo,
 		p = strstr(decode, "-cbc");
 		*p = '\0';
 	}
-	evp = EVP_get_cipherbyname(decode);
+	/*
+	 * Not all versions of libcrypto support calls to add aliases
+	 * to ciphers - newer versions of libressl don't - so, instead
+	 * of making "3des" an alias for "des_ede3_cbc", if attempting
+	 * to get the cipher fails and the name is "3des", we try
+	 * "des_ede3_cbc".
+	 */
+	real_decode = decode;
+	if (strcmp(real_decode, "3des") == 0)
+		real_decode = "des-ede3-cbc";
+	evp = EVP_get_cipherbyname(real_decode);
 
 	if (!evp) {
-		(*ndo->ndo_warning)(ndo, "failed to find cipher algo %s\n", decode);
+		if (decode != real_decode)
+			(*ndo->ndo_warning)(ndo, "failed to find cipher algo %s (%s)\n", real_decode, decode);
+		else
+			(*ndo->ndo_warning)(ndo, "failed to find cipher algo %s\n", decode);
 		sa->evp = NULL;
 		sa->authlen = 0;
 		sa->ivlen = 0;
@@ -684,7 +698,6 @@ static void esp_init(netdissect_options *ndo _U_)
 #if !defined(OPENSSL_API_COMPAT) || OPENSSL_API_COMPAT < 0x10100000L
 	OpenSSL_add_all_algorithms();
 #endif
-	EVP_add_cipher_alias(SN_des_ede3_cbc, "3des");
 }
 DIAG_ON_DEPRECATION
 

